sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3833,9 +3833,10 @@ static size_t imageElementByteSize(CUDA_ARRAY_DESCRIPTOR array_desc) {
   case CU_AD_FORMAT_SIGNED_INT32:
   case CU_AD_FORMAT_FLOAT:
     return 4;
+  default:
```

Comment:
I expect this to emit a warning with CUDA 10.2. :)
Probably the version which can be compiled with both SDKs should process CUDA 11 specific enum values under check `CUDA_VERSION` macro.

Reply:
I agree. When support for the new enum is added, it should be under the constraint `CUDA_VERSION >= 11000`, but until that is implemented I think this is the best solution.

As for the warning, I suppose the `default` case could be wrapped in the `CUDA_VERSION >= 11000` constraint until the new macro is implemented. I think my primary concern is that once the new enum value is implemented (and assuming this does cause a warning), then we either have to remove the `default` case or give it a new constraint if we want to avoid the warning reappearing. If we remove it then we might end up in the same situation in the future. If we update the constraint to a higher (yet unreleased CUDA version), this warning may reappear once such a version is released, assuming it doesn't add a new enum value to `CUDA_ARRAY_DESCRIPTOR `. I can't think of a stable solution to this off the top of my head.

Reply:
Let's merge this version then.

FYI: there are no warnings in CI build with CUDA 10.1 due to `-Wno-covered-switch-default` flag - https://github.com/intel/llvm/blob/sycl/sycl/plugins/CMakeLists.txt#L4.